client/my-sites/checkout/upsell-nudge/index.jsx

```diff
@@ -428,6 +433,13 @@ export class UpsellNudge extends Component {
 		// Professional Email needs to add the locally built cartItem to the cart,
 		// as we need to handle validation failures before redirecting to checkout.
 		if ( PROFESSIONAL_EMAIL_UPSELL === upsellType ) {
+			// Make sure we store the original post-checkout destination in a cookie
+			const destinationFromCookie = retrieveSignupDestination();
+			if ( ! destinationFromCookie ) {
+				const initialThankYouUrl = this.getThankYouPageUrlForIncomingCart( true );
+				persistSignupDestination( initialThankYouUrl );
+			}
+
```

Comment:
Maybe we should the destination only if the `replaceProductsInCart()` call below was successful?